src/h265_nal_unit_parser.cc

```diff
@@ -74,6 +74,20 @@ H265NalUnitHeaderParser::ParseNalUnitHeader(const uint8_t* data,
   return ParseNalUnitHeader(&bit_buffer);
 }
 
+bool H265NalUnitHeaderParser::ParseNalUnitType(const uint8_t* data, const size_t length,
+      NalUnitType& naluType) noexcept {
+  if (length == 0) {
+    return false;
+  }
+  rtc::BitBuffer bitBuffer(data, length);
+  auto naluHeader = ParseNalUnitHeader(&bitBuffer);
+  if (!naluHeader) {
+    return false;
+  }
+  naluType = static_cast<NalUnitType>(naluHeader->nal_unit_type);
```

Comment:
this is not doing anything

Reply:
You mean static_cast is not needed? naluHeader->nal_unit_type is uint32_t, whereas return type is NalUnitType enum, so I had to add static_cast.

Reply:
Oh, ignore the comment. I was thinking about something else

Reply:
Can you please rename the function to `GetNalUnitType()` ?

Reply:
I actually initially had it a GetNalUnitType. But thought that might be a bit confusing because getter functions usually have the return type as the value they are getting. In this case that would mean the return type should be NalUnitType. But if we do that, we won't be able to report parsing error to caller? Like what type would we return if we run into a parsing error?